src/AppBundle/Command/ImportSaleCommand.php

```diff
@@ -83,21 +81,13 @@ private function getSales($limit, $offset)
         return $response->json();
     }
 
-    private function saveSales($jsonSales, $existingInvoices)
+    private function saveSales($jsonSales, $repository)
     {
-        foreach ($jsonSales as $jsonSale) {
-            $exists = false;
-            foreach ($existingInvoices as $existing) {
-                if (
-                    $existing['invoice'] == $jsonSale['invoice'] &&
-                    $existing['licenseId'] == $jsonSale['licenseId'] &&
-                    $existing['pluginKey'] == $jsonSale['pluginKey']
-                ) {
-                    $exists = true;
-                }
-            }
+        $newCnt = 0;
 
-            if (!$exists) {
+        foreach ($jsonSales as $jsonSale) {
+            if ($repository->findIfSaleIsNew($jsonSale['invoice'], $jsonSale['licenseId'], $jsonSale['pluginKey'])) {
```

Comment:
Hm, you've got a valid point.
At worst, it would be enough to simple remove all sales data from the DB.

I will implement this flag later, as well as some indication of counter discrepancy (if total amount in API is different from the DB)